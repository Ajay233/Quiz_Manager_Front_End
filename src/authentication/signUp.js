import React from 'react';
import { connect } from 'react-redux';
import { setNotification } from '../notifications/actions'
import Notification from '../notifications/notifications'
import SignUpForm from '../forms/signUpForm'

import '../stylesheets/inputs.css'
import '../stylesheets/buttons.css'
import '../stylesheets/signup.css'

export class SignUp extends React.Component {

  render(){
    const { setNotification } = this.props
    return(
      <div className="signupContainer">
        <div className="signup">
        <div className="notificationContainer">
          <Notification />
        </div>
          <SignUpForm setNotification={setNotification} />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    notificationData: state.notificationData
  }
}

export default connect(mapStateToProps, {setNotification})(SignUp);
