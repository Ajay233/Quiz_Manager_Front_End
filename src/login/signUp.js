import React from 'react';

class SignUp extends React.Component {
  render(){
    return(
      <div>
        Sign up below
        <form>
          <input></input>
          <input></input>
          <input></input>
          <input></input>
          <button>Go</button>
        </form>
      </div>
    );
  }
}

export default SignUp;
