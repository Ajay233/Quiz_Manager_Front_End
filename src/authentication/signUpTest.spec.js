import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { SignUp, mapStateToProps } from './signUp'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import history from '../history'

const mockStore = configureStore({})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      notificationData: { message: "", type: "", show: false, timed: true }
    }
    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})


describe("SignUp", () => {
  it("should render the signup page and form", () => {
    const store = mockStore({
      notificationData: { message: "", type: "", show: false, timed: true }
    })

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <SignUp globals={{loaderState:{show: false}}}/>
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })

  it("should render the signup page and form and a notification", () => {
    const store = mockStore({
      notificationData: { message: "Test signup worked", type: "success", show: true, timed: true }
    })

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <SignUp globals={{loaderState:{show: false}}}/>
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })
})
