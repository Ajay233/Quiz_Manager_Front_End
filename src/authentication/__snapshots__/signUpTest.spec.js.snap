// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render the signup page and form 1`] = `
<div
  className="signupContainer"
>
  <div
    className="signup"
  >
    <div
      className="notificationContainer"
    />
    <div>
      <form
        className="signupForm"
        onSubmit={[Function]}
      >
        <div
          className="signupTitle"
        >
          Create Account
        </div>
        <div>
          <label>
            First Name:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="forename"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Surname:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="surname"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Email address:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Password:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="submit signupButton"
        >
          Create Account
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`SignUp should render the signup page and form and a notification 1`] = `
<div
  className="signupContainer"
>
  <div
    className="signup"
  >
    <div
      className="notificationContainer"
    >
      <div
        className="spacing success"
        id="notification"
      >
        Test signup worked
      </div>
    </div>
    <div>
      <form
        className="signupForm"
        onSubmit={[Function]}
      >
        <div
          className="signupTitle"
        >
          Create Account
        </div>
        <div>
          <label>
            First Name:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="forename"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Surname:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="surname"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Email address:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type=""
            value=""
          />
        </div>
        <div>
          <label>
            Password:
          </label>
          <input
            autoComplete="off"
            className="inputBox null"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="submit signupButton"
        >
          Create Account
        </button>
      </form>
    </div>
  </div>
</div>
`;
